lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,486 @@
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    REQUESTS_IMP_ERR = traceback.format_exc()
+    HAS_REQUESTS = False
+
+from ansible.errors import AnsibleError, AnsibleFileNotFound
+from ansible.module_utils._text import to_bytes, to_native
+from ansible.plugins.connection import ConnectionBase
+from ansible.utils.path import makedirs_safe
+from ansible.module_utils.basic import missing_required_lib
+
+try:
+    from pyVim.connect import Disconnect, SmartConnect, SmartConnectNoSSL
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+    PYVMOMI_IMP_ERR = traceback.format_exc()
+
+
+DOCUMENTATION = """
+    author: Deric Crago <deric.crago@gmail.com>
+    connection: vmware_tools
+    short_description: Execute tasks inside a VM via VMware Tools
+    description:
+      - Use VMware tools to run tasks in, or put/fetch files to guest operating systems running in VMware infrastructure.
+      - In case of Windows VMs, set C(ansible_shell_type) to C(powershell).
+      - Does not work with 'become'.
+    version_added: "2.8"
+    requirements:
+      - pyvmomi (Python library)
+      - requests (Python library)
+    options:
+      vmware_host:
+        description:
+          - FQDN or IP Address for the connection (vCenter or ESXi Host).
+        env:
+          - name: VI_SERVER
+          - name: VMWARE_HOST
+        vars:
+          - name: ansible_host
+          - name: ansible_vmware_host
+        required: True
+      vmware_user:
+        description:
+          - Username for the connection.
+          - "Requires the following permissions on the VM:
+               - VirtualMachine.GuestOperations.Execute
+               - VirtualMachine.GuestOperations.Modify
+               - VirtualMachine.GuestOperations.Query"
+        env:
+          - name: VI_USERNAME
+          - name: VMWARE_USER
+        vars:
+          - name: ansible_user
+          - name: ansible_vmware_user
+        required: True
+      vmware_password:
+        description:
+          - Password for the connection.
+        env:
+          - name: VI_PASSWORD
+          - name: VMWARE_PASSWORD
+        vars:
+          - name: ansible_password
+          - name: ansible_vmware_password
+        required: True
+      vmware_port:
+        description:
+          - Port for the connection.
+        env:
+          - name: VI_PORTNUMBER
+          - name: VMWARE_PORT
+        vars:
+          - name: ansible_port
+          - name: ansible_vmware_port
+        required: False
+        default: 443
+      validate_certs:
+        description:
+          - Verify SSL for the connection.
+          - "Note: This will validate certs for both C(vmware_host) and the ESXi host running the VM."
+        env:
+          - name: VMWARE_VALIDATE_CERTS
+        vars:
+          - name: ansible_vmware_validate_certs
+        default: True
+        type: bool
+      silence_tls_warnings:
```

Comment:
how are these warnings and not failures? is this controlling behaviour if `validate_certs` is False? would we not just ignore warnings then?

Reply:
It was added by the original author.  @dericcrago is it okay to remove this option and have it default to 'true'?

Reply:
I think we can set it to True by default so that we can just ignore warnings as user set `validate_certs` to `False`

Reply:
Now defaults to `true`